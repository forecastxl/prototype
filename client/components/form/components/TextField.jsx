import React, { PropTypes } from 'react'
import MaterialTextField from 'material-ui/TextField'

function TextField({ input, label, meta: { touched, error }, ...custom }) {
  return (
    <MaterialTextField
      hintText={label}
      floatingLabelText={label}
      errorText={touched && error}
      {...input}
      {...custom}
    />
  )
}

export default TextField
