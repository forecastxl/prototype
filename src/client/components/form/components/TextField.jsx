import React, { PropTypes } from 'react'
import TextField from 'material-ui/TextField'

function TextFieldWrapper({ input, label, meta: { touched, error }, ...custom }) {
  return (
    <TextField
      hintText={label}
      floatingLabelText={label}
      errorText={touched && error}
      {...input}
      {...custom}
    />
  )
}

TextFieldWrapper.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired
}

export default TextFieldWrapper
