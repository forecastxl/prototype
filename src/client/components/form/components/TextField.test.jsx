import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TextField from './TextField'

describe('<TextField />', () => {
  it('renders correctly', () => {
    const meta = { touched: false, error: false }
    const wrapper = shallow(<TextField input={{}} label={'label'} meta={meta} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
