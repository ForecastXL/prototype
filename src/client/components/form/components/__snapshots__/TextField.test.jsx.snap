// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders correctly 1`] = `
<TextField
  disabled={false}
  errorText={false}
  floatingLabelFixed={false}
  floatingLabelText="label"
  fullWidth={false}
  hintText="label"
  multiLine={false}
  rows={1}
  type="text"
  underlineShow={true}
/>
`;

exports[`<TextField /> renders errors 1`] = `
<TextField
  disabled={false}
  errorText="error"
  floatingLabelFixed={false}
  floatingLabelText="label"
  fullWidth={false}
  hintText="label"
  multiLine={false}
  rows={1}
  type="text"
  underlineShow={true}
/>
`;
